File: typescript-openai/test-azure.ts
import dotenv from "dotenv";
import path from "path";
import { fileURLToPath } from "url";
import OpenAI from "openai";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
dotenv.config({ path: path.resolve(__dirname, "../.env") });

const apiKey = process.env.AZURE_OPENAI_API_KEY!;

const client = new OpenAI({
  baseURL: process.env.AZURE_OPENAI_ENDPOINT,
  apiKey,
  defaultHeaders: { "api-key": apiKey },
});

const deployment = process.env.AZURE_DEPLOYMENT_NAME!;

console.log(`Testing Azure OpenAI (streaming) with deployment: ${deployment}`);
console.log("-".repeat(50));

const stream = client.responses.stream({
  model: deployment,
  input: "Say hello and confirm you are running on Azure OpenAI.",
});

for await (const event of stream) {
  if (
    event.type === "response.output_text.delta" &&
    "delta" in event
  ) {
    process.stdout.write(event.delta as string);
  }
}

console.log();

File: python/test_azure.py
import os
from pathlib import Path

from dotenv import load_dotenv
from openai import OpenAI

load_dotenv(Path(__file__).resolve().parent.parent / ".env")

api_key = os.environ["AZURE_OPENAI_API_KEY"]

client = OpenAI(
    base_url=os.environ["AZURE_OPENAI_ENDPOINT"],
    api_key=api_key,
    default_headers={"api-key": api_key},
)

deployment = os.environ["AZURE_DEPLOYMENT_NAME"]

print(f"Testing Azure OpenAI with deployment: {deployment}")
print("-" * 50)

response = client.responses.create(
    model=deployment,
    input="Say hello and confirm you are running on Azure OpenAI.",
)

print(response.output_text)
